### ISSUE TITLE
Question on the function  'validate' of the file 'nnUNetTrainer.py'

### ISSUE THREAD

**Issue by yl255** [MIC-DKFZ/nnUNet#39]
Hi, Fabian
Now I am trying the Cascade model on myself data. I got a problems.
Firstly, I used the 3d_lowres model to train the data. The FOLD is just 4. And it work well.
When I used the 3d_cascade_fullres to train, I got an error: ' seg from prev stage missing: ****' . This error was from the function 'do_split' of the file 'nnUNetTrainerCascadeFullRes.py'.
So I checked the code.
I do not undestand the following code in the function  'validate' of the file 'nnUNetTrainer.py':
   for k in self.dataset_val.keys():
         ......
I think 'dataset_val' should be 'dataset'. So the code is like:
   for k in self.dataset.keys():
         ......
I found 'dataset_val.keys' is just one part of the 'dataset.keys'.

I do not know whether it is a bug? Hope you can understand what I said. :)

Best,
Yaliang 

PS: I found the same code in the function 'predict_next_stage' of the file 'predict_next_stage.py'
          for pat in trainer.dataset_val.keys():
               .......

**Comment by FabianIsensee** [comment]
Hi,
please have a look at https://github.com/MIC-DKFZ/nnUNet#3d-u-net-cascade
You need to run all 5 folds. We can generate the segmentation from the previous stage ONLY for the validation set, otherwise this would be overfitted.
Best,
Fabian
